st2common/st2common/util/monkey_patch.py

```diff
@@ -43,6 +44,23 @@ def monkey_patch(patch_thread=None):
                          patched unless debugger is used.
     :type patch_thread: ``bool``
     """
+    if os.environ.get("ST2_PYCHARM_DEBUG", False):
+
+        # pydevd_pycharm uses this to save copies of stdlib modules before
+        # eventlet monkey patches them, or the debugger will not work.
+        # see: https://intellij-support.jetbrains.com/hc/en-us/community/posts/360000333980
+        os.environ["GEVENT_SUPPORT"] = "True"
+
+        import pydevd_pycharm
+
+        pydevd_pycharm.settrace(
```

Comment:
```suggestion
    if os.environ.get("ST2_DEBUG", False):

        # pydevd_pycharm uses this to save copies of stdlib modules before
        # eventlet monkey patches them, or the debugger will not work.
        # see: https://intellij-support.jetbrains.com/hc/en-us/community/posts/360000333980
        os.environ["GEVENT_SUPPORT"] = "True"

        import pydevd_pycharm

        pydevd_pycharm.settrace(
```

@armab Do you have a recommendation on how we can make this extensible so that people can inject whatever debugger before monkey_patching occurs?